File: app/src/main/java/com/example/dagger_hilt/domain/StorageRepository.java
package com.example.dagger_hilt.domain;

import static com.example.dagger_hilt.sys.util.Constants.PASSWORD;
import static com.example.dagger_hilt.sys.util.Constants.REGISTERED_USER;

import com.example.dagger_hilt.data.datasource.storage.Storage;

import javax.inject.Inject;

public class StorageRepository {

    @Inject
    Storage storage;
    private String userName;

    @Inject
    public StorageRepository() {
    }

    public boolean isUserRegistered() {
        if (storage != null) {
            userName = storage.getString(REGISTERED_USER);
            return !userName.isEmpty();
        } else return false;
    }

    public void registerUser(String userName, String password) {
        storage.setString(REGISTERED_USER, userName);
        storage.setString(userName + PASSWORD, password);
    }


    public boolean loginUser(String userName, String password) {
        isUserRegistered();
        if (!this.userName.equals(userName)) return false;
        var registeredPassword = storage.getString(userName + PASSWORD);
        return registeredPassword.equals(password);
    }

    public void unregister() {
        storage.setString(REGISTERED_USER, "");
        storage.setString(userName + PASSWORD, "");
    }
}

File: app/src/main/java/com/example/dagger_hilt/domain/StorageRepository.kt
package com.example.dagger_hilt.domain

import com.example.dagger_hilt.data.datasource.storage.Storage
import com.example.dagger_hilt.sys.util.Constants.PASSWORD
import com.example.dagger_hilt.sys.util.Constants.REGISTERED_USER
import javax.inject.Inject

class StorageRepository @Inject constructor(private val storage: Storage) {

    private val userName: String
        get() = storage.getString(REGISTERED_USER)

    fun registerUser(userName: String, password: String) {
        storage.setString(REGISTERED_USER, userName)
        storage.setString("$userName$PASSWORD", password)
    }

    fun isUserRegistered() = userName.isNotEmpty()

    fun loginUser(userName: String, password: String): Boolean {
        val registeredUser = this.userName
        if (registeredUser != userName) return false
        val registeredPassword = storage.getString("$userName$PASSWORD")
        if (registeredPassword != password) return false
        return true
    }

    fun unregister() {
        storage.setString(REGISTERED_USER, "")
        storage.setString("$userName$PASSWORD", "")
    }

}